pandas/core/computation/parsing.py

```diff
@@ -102,12 +104,12 @@ def clean_column_name(name: str) -> str:
 
     Parameters
     ----------
-    name : str
+    name : Label
```

Comment:
`Label` -> `hashable`

for the docstring, I think it is still preferred not to use Type notation, even for internal docstrings and follow the documentation contributing guide.